src/App.tsx

```diff
@@ -15,71 +17,64 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+type AppState = {
+  selectedGood: string;
+};
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+export class App extends Component<{}, AppState> {
+  state = { selectedGood: 'Jam' }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {!selectedGood
+          ? <h1 className="title">No goods selected</h1>
+          : (
+            <h1 className="title is-flex is-align-items-center">
+              {`${selectedGood} is selected`}
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => this.setState({ selectedGood: '' })}
+              />
+            </h1>
+          )}
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              const isSelected = good === selectedGood;
+              const buttonClass = cn({ button: true, 'is-info': isSelected });
+              const trClass = cn({
+                'has-background-success-light': isSelected,
+              });
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              return (
+                <tr data-cy="Good" className={trClass} key={good}>
+                  <td>
+                    <button
```

Comment:
This solution is really cool and it seems more complex and ambitious than just rendering a different button based on whether an item is or is not a selected item, although in my opinion it can get a bit hard to read. Also, it forces you to use 3 ternary operators instead of one, so I'm not sure if it's the most optimal solution

A simpler approach would be
`good !== selectedGood
      ? // add button
      : // clear button`